### PR TITLE
Integrated QR Code to share queue link

### PR DESCRIPTION
**Pull request by prabureddy**
Resolves [#430](https://github.com/SimplQ/simplQ-frontend/issues/430)

### CODE REVIEW

File: simplq/src/components/common/Popup/QrCode.jsx [hunk 1]
```diff
@@ -0,0 +1,77 @@
...
+import { useReactToPrint } from 'react-to-print';
+import StandardButton from '../Button';
+
+const ComponentToPrint = forwardRef(({ style, url }, ref) => {
+  return (
+    <div className={style} ref={ref}>
+      <h2>Scan this QR to begin!</h2>
+      <QRCode value={url} />
+      <p>{`or visit ${url}`}</p>
+    </div>
```
Comment (daltonfury42) [inline simplq/src/components/common/Popup/QrCode.jsx:14]: In the generated PDF, the QR code is coming on the top. Is it possible to add some margin / center the printed component?

image

Also, let's add as a title to the modal component (and the PDF) the queue name in bigger font.

**Review by daltonfury42 — CHANGES_REQUESTED** [review]
Nice libraries, good find!

File: simplq/src/components/common/Popup/QrCode.jsx [hunk 2]
```diff
@@ -0,0 +1,77 @@
+import React, { useState, useRef, forwardRef } from 'react';
+import { makeStyles, Modal, useMediaQuery } from '@material-ui/core';
```
[the remark this answers is not in the document]
Comment (prabureddy) [reply simplq/src/components/common/Popup/QrCode.jsx:2]: yes

File: simplq/src/components/common/Popup/QrCode.jsx (hunk 1 above)
Comment (prabureddy) [reply simplq/src/components/common/Popup/QrCode.jsx:14]: Is this Ok?
Screenshot from 2021-01-12 22-12-49
Screenshot from 2021-01-12 22-12-54
I will commit changes and push

Comment (daltonfury42) [reply simplq/src/components/common/Popup/QrCode.jsx]: This is perfect!

Comment (prabureddy) [reply simplq/src/components/common/Popup/QrCode.jsx]: That's good!

File: simplq/src/components/common/utilFns.js [hunk 3]
```diff
@@ -29,3 +29,5 @@ export const smoothScrollToHomePageTop = () => {
     window.location.href = '/';
   }
 };
+
+export const title = (txt) => txt?.charAt(0)?.toUpperCase() + txt?.substr(1)?.toLowerCase();
```
Comment (daltonfury42) [inline simplq/src/components/common/utilFns.js:33]: https://stackoverflow.com/questions/7225407/convert-camelcasetext-to-sentence-case-text

**Review by daltonfury42 — APPROVED** [review]

**Comment by maaverik** [comment]
This looks fantastic. Awesome job @prabureddy.

**Comment by prabureddy** [comment]
> This looks fantastic. Awesome job @prabureddy.

Thanks. I'd love to work on this repository whenever I'm free.

**Pull request merged**